File: wear-os/app/src/main/java/com/grantlittman/wearapp/data/model/CountdownWarning.kt
package com.grantlittman.wearapp.data.model

/**
 * A distinct signal pattern that activates during the final stretch of a session.
 *
 * @param activateAtMillis How many milliseconds before the end to start the countdown
 *                         (e.g., 120_000 = activate with 2 minutes remaining).
 * @param intervalMillis How often to fire during the countdown (e.g., 15_000 = every 15 seconds).
 * @param signal The signal to produce during the countdown.
 */
data class CountdownWarning(
    val activateAtMillis: Long,
    val intervalMillis: Long,
    val signal: Signal
)

File: watchos/PulseTimer/PulseTimer Watch App/Models/CountdownWarning.swift
import Foundation

/// A distinct signal pattern that activates during the final stretch of a session.
///
/// `activateAtMillis` is the number of milliseconds *before* the end of the session
/// to begin the countdown (e.g., 120_000 = activate with 2 minutes remaining).
struct CountdownWarning: Codable, Equatable, Hashable {
    var activateAtMillis: Int64
    var intervalMillis: Int64
    var signal: Signal
}
